### PR TITLE
AV-37800 Add support for crypto object mapping in AVI config converter.

### PR DESCRIPTION
**Pull request by gitshrikant**
Added support for crypto chain-group object and successfully uploaded configuration to Avi controller.

### CODE REVIEW

File: python/avi/migrationtools/ace_converter/ace_converter.py [hunk 1]
```diff
@@ -225,7 +224,7 @@ def print_pip_and_controller_version(self):
                         help='location of key and cert file')
 
     parser.add_argument('-o', '--output_loc',
-                        help='Out file location')
+                        help='Out file location',default='.')
```
Comment (grastogi23) [inline python/avi/migrationtools/ace_converter/ace_converter.py]: there should be space before deafult.

File: python/avi/migrationtools/ace_converter/ace_parser.py [hunk 2]
```diff
@@ -493,7 +508,7 @@ def parse_ace(self):
             name_to_log = None
             type_to_log = ['logging', 'access-list', 'rserver', 'serverfarm',
                            'parameter-map', 'class-map', 'policy-map', 'sticky',
-                           'probe', 'action-list']
+                           'probe', 'action-list','crypto']
```
Comment (grastogi23) [inline python/avi/migrationtools/ace_converter/ace_parser.py]: space after comma

File: python/avi/migrationtools/ace_converter/ace_parser.py [hunk 3]
```diff
@@ -575,12 +590,22 @@ def parse_ace(self):
...
+                                match[0]: match[1],
+                                'port': match[2],
+                                'enabled': match[3]
+                            }
+                        else:
+                            temp_dict = {
+                                match[0]: match[1],
+                                'enabled': match[2]
+                            }
+                    if len(temp_dict.keys()) > 1:
```
Comment (grastogi23) [inline python/avi/migrationtools/ace_converter/ace_parser.py]: if temp_dict:

File: python/avi/migrationtools/ace_converter/ace_parser.py [hunk 4]
```diff
@@ -575,12 +590,22 @@ def parse_ace(self):
                         temp_dict = {
                             match[0]: match[1]
                         }
-                    else:
-                        temp_dict = {
-                            match[0]: match[1],
-                            'type': match[2]
-                        }
-                    extra_dict['desc'].append(temp_dict)
+                    elif 'rserver' in match:
```
Comment (grastogi23) [inline python/avi/migrationtools/ace_converter/ace_parser.py]: add comments on significance of 3, 4, etc. give an example of how it is breaking up .

File: python/avi/migrationtools/ace_converter/ace_parser.py [hunk 5]
```diff
@@ -785,6 +816,33 @@ def parse_ace(self):
                     extra_dict.update(temp_dict)
                 LOG.info('parsing: probe for value : {}'.format(name_to_log))
 
+            if key == 'crypto':
```
Comment (grastogi23) [inline python/avi/migrationtools/ace_converter/ace_parser.py]: let us break up into smaller code chunks.

**Review by grastogi23 — CHANGES_REQUESTED** [review]

**Review by gitshrikant — COMMENTED** [review]
Done with the suggested changes.

File: python/avi/migrationtools/ace_converter/ace_parser.py [hunk 6]
```diff
@@ -113,7 +109,7 @@ def parse_ace(self):
         rserver_name = Word(printables)
 
         grammer_4_1 = Group(rserver_key + host + rserver_name)
-        grammer_4_2 = Group(Keyword('description') + restOfLine)
```
Comment (chaitanyaavi) [inline python/avi/migrationtools/ace_converter/ace_parser.py]: Should be restOfLine only test with description having blank spaces

**Review by grastogi23 — APPROVED** [review]

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**